### PR TITLE
Fix JENKINS-62444

### PR DESCRIPTION
**Pull request by slide**
This updates email-ext to be compliant with the JCaSC plugin and includes a test for support. Thanks to @alecharp for his help!

### CODE REVIEW

**Comment by slide** [comment]
I have a fix for the build issue, will push it soon

**Comment by alecharp** [comment]
@slide could we cut a release of the plugin with this?

**Comment by slide** [comment]
Yes, I am planning on doing that today.

**Comment by slide** [comment]
The release is out now, had issues with my release machine.

File: src/main/java/hudson/plugins/emailext/ExtendedEmailPublisherDescriptor.java [hunk 1]
```diff
@@ -54,6 +54,7 @@
  * These settings are global configurations
  */
 @Extension
+@Symbol("email-ext")
```
Comment (jglick) [inline src/main/java/hudson/plugins/emailext/ExtendedEmailPublisherDescriptor.java:57]: This is not backward-compatible unfortunately.

Comment (jglick) [reply src/main/java/hudson/plugins/emailext/ExtendedEmailPublisherDescriptor.java:57]: I would suggest

```java
@Symbol({"email-ext", "extendedEmailPublisher"})
```

if CasC in fact behaves properly: accepting either, but suggesting only the first. @timja @jetersen et al.?

Comment (timja) [reply src/main/java/hudson/plugins/emailext/ExtendedEmailPublisherDescriptor.java:57]: Yes casc works with that, anyone using the previous behaviour will also prompt an administrative warning telling them to update to the new syntax:

e.g. (on my instance which I haven't got around to fixing)

image

Comment (batmat) [reply src/main/java/hudson/plugins/emailext/ExtendedEmailPublisherDescriptor.java:57]: I'm unclear whether this is actually breaking compatibility, or not. IOW, is this supposed to be only a warning, or an instance using the previous values would not start without adapting their yaml (or wait for https://github.com/jenkinsci/email-ext-plugin/pull/271)

Comment (timja) [reply src/main/java/hudson/plugins/emailext/ExtendedEmailPublisherDescriptor.java:57]: it depends on whether someone was using the previous symbol for something or whether any jcasc was possible before without this

**Pull request merged**